tools/install_demo_configuration.bat

```diff
@@ -319,6 +321,66 @@ echo plugins.security.restapi.roles_enabled: ["all_access", "security_rest_api_a
 echo plugins.security.system_indices.enabled: true >> "%OPENSEARCH_CONF_FILE%"
 echo plugins.security.system_indices.indices: [".plugins-ml-config", ".plugins-ml-connector", ".plugins-ml-model-group", ".plugins-ml-model", ".plugins-ml-task", ".plugins-ml-conversation-meta", ".plugins-ml-conversation-interactions", ".opendistro-alerting-config", ".opendistro-alerting-alert*", ".opendistro-anomaly-results*", ".opendistro-anomaly-detector*", ".opendistro-anomaly-checkpoints", ".opendistro-anomaly-detection-state", ".opendistro-reports-*", ".opensearch-notifications-*", ".opensearch-notebooks", ".opensearch-observability", ".ql-datasources", ".opendistro-asynchronous-search-response*", ".replication-metadata-store", ".opensearch-knn-models", ".geospatial-ip2geo-data*", ".opendistro-job-scheduler-lock"] >> "%OPENSEARCH_CONF_FILE%"
 
+
+REM Initialize the variable
+set "ADMIN_PASSWORD="
+
+REM Read the content of admin_password.txt into the ADMIN_PASSWORD variable
+for /f "usebackq" %%i in ("%ADMIN_PASSWORD_FILE%") do (
+    set "ADMIN_PASSWORD=%%i"
+)
+
+REM If ADMIN_PASSWORD is empty, check the environment variable as a fallback
+if not defined ADMIN_PASSWORD (
+    if defined ENV_ADMIN_PASSWORD (
+        set "ADMIN_PASSWORD=!ENV_ADMIN_PASSWORD!"
+    ) else (
+        echo Unable to find admin password for cluster, please run "set ENV_ADMIN_PASSWORD=<your_password>" or create a file {OPENSEARCH_ROOT}\admin_password.txt with a single line that contains the password followed by a newline.
+        exit /b 1
+    )
+)
+
+
+set "salt="
+for /l %%i in (1,1,16) do (
+    set /a "rand=!random! %% 16"
+    set "salt=!salt!!rand!"
+)
+
+openssl passwd -bcrypt -salt !salt! "!ADMIN_PASSWORD!" > tmp_hash.txt
```

Comment:
Doesn't work on my machine, `passwd: Option unknown option -bcrypt`

```
C:\Users\petern> openssl version -a
OpenSSL 1.1.1p  21 Jun 2022
built on: Tue Jun 21 20:02:49 2022 UTC
platform: mingw64
options:  bn(64,64) rc4(16x,int) des(long) idea(int) blowfish(ptr)
compiler: gcc -m64 -Wall -O3 -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DUNICODE -D_UNICODE -DWIN32_LEAN_AND_MEAN -D_MT -DZLIB -DZLIB_SHARED -DNDEBUG -D__MINGW_USE_VC2005_COMPAT -DOPENSSLBIN="\"/mingw64/bin\""
OPENSSLDIR: "/mingw64/ssl"
ENGINESDIR: "/mingw64/lib/engines-1_1"
Seeding source: os-specific
```

Reply:
Will use the password hash tool instead.